sstables/mx/reader.cc

```diff
@@ -1511,10 +1511,16 @@ class mx_sstable_mutation_reader : public mp_row_consumer_reader_mx {
             }
         });
     }
+    bool has_sstable_attached() const noexcept {
+        return bool(_sst);
+    }
     bool is_initialized() const {
         return bool(_context);
     }
     future<> initialize() {
+        if (!has_sstable_attached()) {
+            co_return;
+        }
```

Comment:
Please add comments here, it looks odd.

Another option is for the seek process to determine there's nothing in the range, and return an empty reader.

Reply:
Or maybe seek to the very end of the sstable and let it naturally terminate.

Reply:
If we enabled fast forward, combined reader cannot tell underlying readers are really exhausted.

Think that we're cleaning up a SSTable run, where each fragment overlaps a different owned range

When reading the first owned range, combined reader will call incremental selector and fetch the first fragment.

Once we're done exhausting it, underlying reader returns EOS

but is the reader really exhausted (I am asking this from combined reader's perspective)? It may have data for the next range we'll fast forward to, so it remains opened and listed by combined reader.

On the next fast forward call, the EOS flag of the underlying reader is cleared and the reader is asked again for data

Only the underlying reader itself knows that it has been exhausted, as it's the one that has the notion of EOF. 

The flat reader interface only knows about exhaustion of a reader for a particular range

Reply:
I see. Then a mutation_source really is the best approach.

We could have hidden it in a layered reader, but it's also good to avoid adding cycles to the data path.